omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ConditionController2_2Test.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Tests functionality of {@link ConditionController}.
+ */
+public class ConditionController2_2Test extends MainResourceControllerTest {
+	
+	private ConditionService conditionService;
+	
+	private Patient patient;
+	
+	@Before
+	public void before() throws Exception {
+		this.conditionService = Context.getConditionService();
+		executeDataSet(RestTestConstants2_2.CONDITION_TEST_DATA_XML);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return "condition";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.CONDITION_UUID;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return conditionService.getActiveConditions(this.patient).size();
+	}
+	
+	@Test
+	public void shouldCreateACondition() throws Exception {
+		long originalCount = getAllCount();
+		
+		SimpleObject conditionSource = new SimpleObject();
+		conditionSource.add("condition", 2);
+		conditionSource.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		conditionSource.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		conditionSource.add("onsetDate", "2018-02-20 00:00:00.0");
+		
+		String json = new ObjectMapper().writeValueAsString(conditionSource);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newConditionSource = deserialize(handle(req));
+		
+		Assert.assertNotNull(PropertyUtils.getProperty(newConditionSource, "uuid"));
+		Assert.assertEquals(originalCount + 1, getAllCount());
+
+		Condition condition = conditionService.getConditionByUuid(getUuid());
+		Assert.assertEquals(newConditionSource.get("condition"), condition.getConditionId());
+		Assert.assertEquals(newConditionSource.get("clinicalStatus"), condition.getClinicalStatus());
+		Assert.assertEquals(newConditionSource.get("verificationStatus"), condition.getVerificationStatus());
+		Assert.assertEquals(newConditionSource.get("onsetDate"), condition.getOnsetDate());
+	}
+
+	@Test
+	public void voidCondition_shouldVoidACondition() throws Exception {
+		Condition condition = conditionService.getConditionByUuid(getUuid());
+		Assert.assertFalse(condition.isVoided());
+
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", "unit test");
+		handle(req);
+
+		condition = conditionService.getConditionByUuid(getUuid());
+		Assert.assertTrue(condition.isVoided());
+		Assert.assertEquals("unit test", condition.getVoidReason());
+	}
+
+	@Test
+	public void unvoidCondition_shouldUnvoidACondition() throws Exception {
+
+		Condition condition = conditionService.getConditionByUuid(getUuid());
```

Comment:
How can this same condition be voided here and then not voided in the previous test?